soql-pack/src/main/scala/com.socrata.soql/SoQLPackDecoder.scala

```diff
@@ -80,6 +81,18 @@ object SoQLPackDecoder {
   def decodeBoolean(item: Any): Option[SoQLValue] =
     Some(SoQLBoolean(item.asInstanceOf[Boolean]))
 
+  def decodeUrl(item: Any): Option[SoQLValue] = item match {
+    case Seq(url: Array[Byte], label: Array[Byte]) =>
+      Some(SoQLUrl(Some(new String(url, StandardCharsets.UTF_8)), Some(new String(label, StandardCharsets.UTF_8))))
+    case Seq(url: Array[Byte], null) =>
+      Some(SoQLUrl(Some(new String(url, StandardCharsets.UTF_8)), None))
+    case Seq(null, label: Array[Byte]) =>
+      Some(SoQLUrl(None, Some(new String(label, StandardCharsets.UTF_8))))
+    case Seq(null, null) =>
+      Some(SoQLUrl(None, None))
```

Comment:
Annoyingly, this is probably the most concise way to write this.

Reply:
wait why not return a None on ln 92?

Reply:
or at least have a catch-all